src/libcadet/model/StirredTankModelVarPor.cpp

```diff
@@ -0,0 +1,2121 @@
+	if (paramProvider.exists("REACTION_MODEL"))
+	{
+		const std::string dynReactName = paramProvider.getString("REACTION_MODEL");
+		_dynReactionBulk = helper.createDynamicReactionModel(dynReactName);
+		if (!_dynReactionBulk)
+			throw InvalidParameterException("Unknown dynamic reaction model " + dynReactName);
+
+		if (_dynReactionBulk->usesParamProviderInDiscretizationConfig())
+			paramProvider.pushScope("reaction_bulk");
+
+		reactionConfSuccess = _dynReactionBulk->configureModelDiscretization(paramProvider, _nComp, nullptr, nullptr);
+
+		if (_dynReactionBulk->usesParamProviderInDiscretizationConfig())
+			paramProvider.popScope();
+	}
+
+	_dynReaction = std::vector<IDynamicReactionModel*>(_nParType, nullptr);
+
+	if (paramProvider.exists("REACTION_MODEL_PARTICLES"))
+	{
+		const std::vector<std::string> dynReactModelNames = paramProvider.getStringArray("REACTION_MODEL_PARTICLES");
+		if (dynReactModelNames.size() < _nParType)
+			throw InvalidParameterException("Field REACTION_MODEL_PARTICLES contains too few elements (" + std::to_string(_nParType) + " required)");
+
+		for (unsigned int i = 0; i < _nParType; ++i)
+		{
+			_dynReaction[i] = helper.createDynamicReactionModel(dynReactModelNames[i]);
+			if (!_dynReaction[i])
+				throw InvalidParameterException("Unknown dynamic reaction model " + dynReactModelNames[i]);
+
+			MultiplexedScopeSelector scopeGuard(paramProvider, "reaction_particle", _nParType == 1, i, _nParType == 1, _dynReaction[i]->usesParamProviderInDiscretizationConfig());
+			reactionConfSuccess = _dynReaction[i]->configureModelDiscretization(paramProvider, _nComp, _nBound + i * _nComp, _boundOffset + i * _nComp) && reactionConfSuccess;
+		}
+	}
+
+	return bindingConfSuccess && reactionConfSuccess;
+}
+
+bool CSTRModelVarPor::configure(IParameterProvider& paramProvider)
+{
+	_curFlowRateFilter = 0.0;
+	_flowRateFilter.clear();
+	const bool hasFlowrateFilter = paramProvider.exists("FLOWRATE_FILTER");
+	if (hasFlowrateFilter)
+		readScalarParameterOrArray(_flowRateFilter, paramProvider, "FLOWRATE_FILTER", 1);
+
+	_constSolidVolume = 0.0;
+	if (paramProvider.exists("CONST_SOLID_VOLUME"))
+		_constSolidVolume = paramProvider.getDouble("CONST_SOLID_VOLUME");
+	else if (paramProvider.exists("POROSITY")) {
+		
+		// std::cerr << "Warning: CSTR don't support POROSITY. CADET is trying to set CONST_SOLID_VOLUME = INIT_VOLUME / POROSITY - INIT_VOLUME." << std::endl;
+		// std::cerr << "Please press ENTER to continue. " << std::endl;
+		// AB std::cin.get(); 
+		throw InvalidParameterException("CSTR don't support POROSITY. Pleas set CONST_SOLID_VOLUME = INIT_VOLUME / POROSITY - INIT_VOLUME." );
+	}
+
+	if (paramProvider.exists("POROSITY") && paramProvider.exists("CONST_SOLID_VOLUME")){
+
+		const double eps = paramProvider.exists("POROSITY");
+		const double sVolume = paramProvider.exists("CONST_SOLID_VOLUME");
+		const double initVolume = paramProvider.getDouble("INIT_VOLUME");
+
+		if (sVolume != initVolume / eps - initVolume)
```

Comment:
This is an exact floating point comparison. Due to round-off and / or representation errors (e.g., `0.3` cannot be represented exactly), it may not work. Allow some slack by introducing relative and / or absolute tolerance: https://codingnest.com/the-little-things-comparing-floating-point-numbers/